src/test/ui/suggestions/issue-81839.stderr

```diff
@@ -17,7 +17,7 @@ LL | |     }
   ::: $DIR/auxiliary/issue-81839.rs:6:49
    |
 LL |       pub async fn answer_str(&self, _s: &str) -> Test {
-   |                                                   ---- checked the `Output` of this `async fn`, found opaque type
+   |                                                   ---- calling an async function returns a future
```

Comment:
This looks like the diagnostic is saying that `Test` is a future. Could we instead say something like 'an async function returns a future that outputs the return type'?

Reply:
Maybe we could highlight the entire function signature and say something like:
```
this async function will return an `impl Future<Output = Test>`
```

Reply:
To better support macros, I think we should keep pointing to the return type, but mention the function name in the error message. This will produce a nicer message when the return type tokens are constructed separately (e.g. a type passed into a `macro_rules!` macro).

Reply:
Thank you for the suggestion - any pointers on how to surface the function name in the error message?

Reply:
It doesn't look conveniently accessible at the moment. Maybe @estebank has a suggestion on the cleanest way to get the name of the desugared async function?

Reply:
In https://github.com/rust-lang/rust/pull/86705/files#diff-845d26194de711d60e1317dbe90812d5d7635f4f758222fd7e14bf550ffe895dL1522 we have the `DefId` of the opaque type. With it I think you can get the def id of the parent `fn`, and call `def_path` on _that_, but I don't have the codebase at hand now to be more specific. I can help you nell either tomorrow or on thursday.